core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java

```diff
@@ -156,7 +156,9 @@ public static void run(Application application, Class<? extends QuarkusApplicati
                 }
             }
         } catch (Exception e) {
-            Throwable rootCause = ExceptionUtil.getRootCause(e);
+            // Here it is trying to get rootCause in which when ConfigurationException is thrown
+            // is ConfigValidationException from io.smallrye.config
+            Throwable rootCause = e instanceof ConfigurationException ? e : ExceptionUtil.getRootCause(e);
```

Comment:
Do you know what wraps the exception in a `RuntimeException` exactly? It might be a better idea to just remove that wrapping, if possible. At least for `ConfigurationException`.

Reply:
Actually, from what I saw you did avoid the wrapping... So why do this? :thinking:

Reply:
Why do I check here if it is an instance of ConfigurationException? It is because If I don't do this it was getting rootCause by default which in this case was ConfigValidationException from io.smallrye.config
Even when the wrapped ConfigurationException was catched here, it would still go into the else { } block due to this reason

Reply:
Right, my bad :facepalm: Shouldn't have reviewed this first thing in the morning... Thanks for the explanation.